### PR TITLE
Add marker to reboot test

### PR DESCRIPTION
**Pull request by JanneKiiskila**
Multiple changes:
* Add reboot marker.
* Move `pytest.ini` to root to get the customer marker really read.
* Update `pytest` and `pytest-html` plugin to resolve Python 3.10 compatibility issue.
* Add `*.xml` -files to `.gitignore`.

More on the `reset_marker`.

This allows you to run the test suite with -m option to exclude the reboot test in case you don't want to do it.

```
pytest -m "not reboot_test"
```

This shows the following output then to indicate that test is skipped.

```
(venv)  13:53 jannek@jannek-P720:/ssd/e2e-edge-test-suite$ pytest  --config_path=config-os2-snap20.json -m "not reboot_test" ====================================================================================================== test session starts ====================================================================================================== platform linux -- Python 3.10.6, pytest-7.2.1, pluggy-0.13.1 rootdir: /ssd/e2e-edge-test-suite
plugins: html-1.22.0, stress-1.0.1, rerunfailures-10.2, repeat-0.9.1, metadata-2.0.2 collected 14 items / 1 deselected / 13 selected
```

I.e. that one case gets deselected.

# Pull request template

Please fill this pull request template to describe your pull request.
If it introduces API breaks please describe it in detail.

### Testing

 - [x] tests run against live device.

 ### Changelog
 
 - [x] `CHANGELOG.md` updated.

 ### Tagging
 
 - [ ] Remember to tag a new release, if need be.

### CODE REVIEW

File: tests/pytest.ini [hunk 1]
```diff
@@ -11,4 +11,6 @@ log_file_level = DEBUG
 log_file_date_format = %Y-%m-%d %H:%M:%S
 log_file_format = %(asctime)s.%(msecs)03d   %(levelname)-8s %(name)s %(message)s
 junit_family = xunit1
-junit_logging = all
\ No newline at end of file
+junit_logging = all
+markers = 
+    reboot_test: this test will reboot device (deselect with '-m "not slow"')
```
Comment (petedyerarm) [inline tests/pytest.ini:16]:
```suggestion
    reboot_test: this test will reboot device (deselect with '-m "not reboot_test"')
```

**Review by petedyerarm — CHANGES_REQUESTED** [review]

**Review by tulsivanol — APPROVED** [review]

File: tests/pytest.ini (hunk 1 above)
Comment (JanneKiiskila) [reply tests/pytest.ini]: Ah, good catch. Copy pasta.
Fixed it.

**Review by petedyerarm — APPROVED** [review]
LGTM

**Comment by JanneKiiskila** [comment]
Still giving me this:

```
======================================================================================================= warnings summary ========================================================================================================
tests/test_websocket_notifications.py:61
  /ssd/e2e-edge-test-suite/tests/test_websocket_notifications.py:61: PytestUnknownMarkWarning: Unknown pytest.mark.reboot_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.reboot_test
```

though according to  [this](https://docs.pytest.org/en/stable/how-to/mark.html#registering-marks): the `pytest.ini` change should be enough.

**Comment by JanneKiiskila** [comment]
Additional commits, lots of rebasing/tidying up to make history more sensible. Please re-review.

**Review by petedyerarm — APPROVED** [review]
LGTM

**Pull request merged**